### PR TITLE
tls: Add test vectors for TLS spec

### LINKED ISSUE #459: Include test vectors in specifications
Whilst reviewing the `rust-libp2p` implementation of the QUIC transport, I took a closer look at the TLS specification: https://github.com/libp2p/specs/blob/master/tls/tls.md

What do people think about including test vectors in there? I am thinking something like:

- Given ed25519 identity keypair X
- Given certificate keypair
- A valid certificate with the custom libp2p protocol extension looks like XYZ

This would make it possible to easily write unit tests against this fairly critical part of the code and would potentially catch interoperability issues early on.

### PR DESCRIPTION
**Pull request by thomaseizinger**
cc @marten-seemann 

Fixes #459.

### CODE REVIEW

**Review by marten-seemann — APPROVED** [review]
LGTM. Test added in https://github.com/libp2p/go-libp2p/pull/1788.

File: tls/tls.md [hunk 1]
```diff
@@ -119,6 +120,50 @@ How the public key is encoded into the `Data` bytes depends on the Key Type.
 - `Secp256k1`: Only the compressed form of the public key. 33 bytes.
 - The rest of the keys are encoded as a [SubjectPublicKeyInfo structure](https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1) in PKIX, ASN.1 DER form.
 
+## Test vectors
+
+The following items presents test vectors that a compatible implementation should pass.
```
Comment (mxinden) [inline tls/tls.md:125]:
```suggestion
The following items present test vectors that a compatible implementation should pass.
```

File: tls/tls.md [hunk 2]
```diff
@@ -119,6 +120,50 @@ How the public key is encoded into the `Data` bytes depends on the Key Type.
...
+Due to the randomness required when signing certificates, it is hard to provide testcases for generating certificates.
+These test cases verify that implements can correctly parse certificates with all key types.
```
Comment (mxinden) [inline tls/tls.md:127]:
```suggestion
These test cases verify that implementations can correctly parse certificates with all key types.
```

**Review by mxinden — APPROVED** [review]
Thanks @thomaseizinger!

Pinging maintainers of other implementations, may be helpful for them as well. @Menduist @Nashatyrev @btoms20 @achingbrain

**Pull request merged**